src/Voron/Data/BTrees/RecentlyFoundTreePages.cs

```diff
@@ -5,145 +5,219 @@
 // -----------------------------------------------------------------------
 
 using System;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
 using Sparrow.Server;
 
 namespace Voron.Data.BTrees
 {
-    public sealed class RecentlyFoundTreePages
+    [StructLayout(LayoutKind.Sequential)]
+    public unsafe struct FoundTreePageDescriptor
     {
-        public sealed class FoundTreePage : IDisposable
-        {
-            public readonly long Number;
-            public readonly Slice FirstKey;
-            public readonly Slice LastKey;
-            public readonly long[] CursorPath;
-            private ByteStringContext.Scope _firstScope;
-            private ByteStringContext.Scope _lastScope;
+        public const int MaxCursorPath = 10;
 
-            public TreePage Page;
+        public TreePage Page;
+        public long Number;
 
-            public FoundTreePage(long number, TreePage page, Slice firstKey, Slice lastKey, long[] cursorPath, ByteStringContext.Scope firstScope, ByteStringContext.Scope lastScope)
-            {
-                Number = number;
-                Page = page;
-                FirstKey = firstKey;
-                LastKey = lastKey;
-                CursorPath = cursorPath;
-                _firstScope = firstScope;
-                _lastScope = lastScope;
-            }
+        public Slice FirstKey;
+        public ByteStringContext.Scope FirstScope;
 
-            public void Dispose()
-            {
-                _firstScope.Dispose();
-               _lastScope.Dispose();
```

Comment:
Who is going to dispose those scopes now? I don't see any `FirstScope.Dispose()` calls. How does it work in current impl?

Reply:
There is no need now, they are embedded into the object now (which is massive).